database/mysql/mysql_db.py

```diff
@@ -338,8 +340,19 @@ def main():
                     changed = db_create(cursor, db, encoding, collation)
                 except Exception, e:
                     module.fail_json(msg="error creating database: " + str(e))
+            module.exit_json(changed=changed, db=db)
+
+        elif state == "import":
+            if module.check_mode:
+                changed = True
+            else:
+                try:
+                    changed = db_create(cursor, db, encoding, collation)
+                except Exception, e:
+                    module.fail_json(msg="error creating database: " + str(e))
 
-    module.exit_json(changed=changed, db=db)
+            module.exit_json(changed=changed, db=db)
+        module.fail_json(msg="Database not found.")
```

Comment:
We should never reach this line (we will, because the module is incomplete). 

Currently, the following combinations will fail
DB exists, state=present - should return changed=False, db=db
DB does not exist, state=dump - should fail with a helpful message like "Cannot dump database DBNAME - not found"
DB does not exist, state=absent - should return changed=False, db=db

If you remove this line, we won't be in a worse state than currently.